### PR TITLE
tests: fix nested errors uc20

### PR DESCRIPTION
**Pull request by sergiocazzolato**
These changes are needed to fix the following issue. With those changes it is not needed to have either qemu 5.0 or a specific kernel.

KVM: entry failed, hardware error 0x80000021

If you're running a guest on an Intel machine without unrestricted mode support, the failure can be most likely due to the guest entering an invalid state for Intel VT. For example, the guest maybe running in big real mode which is not supported on less recent Intel processors.

### CODE REVIEW

File: tests/lib/nested.sh [hunk 1]
```diff
@@ -1142,9 +1110,6 @@ nested_start_core_vm_unit() {
     # wait for the $NESTED_VM service to appear active
     wait_for_service "$NESTED_VM"
 
-    # make sure the service started and it is running
-    nested_retry_start_with_boot_errors
```
Comment (mardy) [inline tests/lib/nested.sh:1146]: I think you should explain a bit better (maybe in the commit message) why this is being removed. It sounds like a good step to have.

If the reason is that it doesn't work on intel machines, maybe you can just modify the nested_retry_start_with_boot_errors() function to have a check on `cpuinfo` and return if it's an Intel?

Comment (sergiocazzolato) [reply tests/lib/nested.sh:1146]: Thanks for the review @mardy 

This function is a bit confusing, when I added that I promised to remove it once the error is gone, with the change that I did in this pr the error is almost not reproduced but it is still happening. 

Now, the error is not happening when we boot the machine, it happens after a while, so this check is not working anymore, I'll change the approach and I'll create a service to monitor the nested-vm and restart it in case this error occurs at any moment. My idea is to create a different pr for that. Does it make sense?

Comment (sergiocazzolato) [reply tests/lib/nested.sh:1146]: I'll try to push this new service in the next commit so we fix the issue now.

Comment (sergiocazzolato) [reply tests/lib/nested.sh:1146]: @mardy I already included a new solution for that. please take a second look

File: tests/lib/nested.sh [hunk 2]
```diff
@@ -651,6 +619,13 @@ defaults:
     journal:
       persistent: true
 EOF
+            if [ "$NESTED_SNAPD_DEBUG_TO_SERIAL" = "true" ]; then
```
Comment (Meulengracht) [inline tests/lib/nested.sh:622]: this is an really nice option to have

File: tests/lib/nested.sh [hunk 3]
```diff
@@ -1114,6 +1090,14 @@ nested_start_core_vm_unit() {
     fi
     PARAM_IMAGE="$PARAM_IMAGE,${PARAM_PHYS_BLOCK_SIZE},${PARAM_LOGI_BLOCK_SIZE}"
 
+    if nested_is_core_20_system; then
+        # This is to deal with the following qemu error that it just happens using q35 machines in focal
```
Comment (Meulengracht) [inline tests/lib/nested.sh:1094]: Maybe reword this a bit, how about
`This is to deal with the following qemu error which occurs using q35 machines in focal`

Comment (sergiocazzolato) [reply tests/lib/nested.sh:1094]: done, thanks for the review
